### PR TITLE
fix many issues with link, add pure java version

### PR DESCRIPTION
**Pull request by jafu888**

### CODE REVIEW

File: desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+                    DefaultMutableTreeNode node = new DefaultMutableTreeNode(name);
+
+                    root.add(node);
+                    i++;
+                }
+                layoutListTree.setRootVisible(false);
+                layoutListTree.setModel(model);
+                break;
+            case MOTION_SCENE_UPDATE:
+
```
Comment (jswong65) [inline desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java:277]: Do we need try catch here as what we have in Main.kt?

File: desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+        model.reload();
+        layoutListTree.expandRow(motionLink.mSelectedIndex);
+    }
+
+    File myTmpFile = null;
+    long myTempLastModified = 0;
+    Timer myTmpTimer = null;
+
+    void remoteEditStop() {
+        myTmpTimer.stop();
```
Comment (jswong65) [inline desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java:345]: Do we need to check if myTmpTimer is null?

File: desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+        myTmpFile.deleteOnExit();
+        myTmpFile = null;
+    }
+
+    public void addDesign(String type) throws CLParsingException, BadLocationException {
```
[the remark this answers is not in the document]
Comment (jafu888) [reply desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/Main2.java:350]: addDesign does not override anything?

File: desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/LayoutEditor2.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+            drawImage(g, 0, y);
+        }
+
+        void drawImage(Graphics2D g, int x, int y) {
+            int h = img.getHeight();
+            int w = img.getWidth();
+            g.drawImage(img, -gap - w, y - h / 2, null);
+        }
+
+        void addToPicker(ScenePicker scenePicker, int ox, int oy, int w, int h) {
```
[the remark this answers is not in the document]
Comment (jafu888) [reply desktop/ConstraintLayoutInspector/app/src/androidx/constraintLayout/desktop/link/java/LayoutEditor2.java:261]: This does not override anything

**Pull request merged**